daly-bms-uart.cpp

```diff
@@ -39,63 +40,277 @@ bool Daly_BMS_UART::Init()
     }
+    return true;
+}
+
+bool Daly_BMS_UART::getPackMeasurements() //0x90
+{
+    this->sendCommand(COMMAND::VOUT_IOUT_SOC);
 
+    if (!this->receiveBytes())
+    {
+#ifdef DALY_BMS_DEBUG
+        DEBUG_SERIAL.print("<DALY-BMS DEBUG> Receive failed, V, I, & SOC values won't be modified!\n");
+#endif
+        return false;
+    }
     // Pull the relevent values out of the buffer
-    voltage = (float)((this->my_rxBuffer[4] << 8) | this->my_rxBuffer[5]) / 10;
+    get.packVoltage = (float)((this->my_rxBuffer[4] << 8) | this->my_rxBuffer[5]) / 10;
     // The current measurement is given with a 30000 unit offset
-    current = (float)(((this->my_rxBuffer[8] << 8) | this->my_rxBuffer[9]) - 30000) / 10;
-    SOC = (float)((this->my_rxBuffer[10] << 8) | this->my_rxBuffer[11]) / 10;
+    get.packCurrent = (float)(((this->my_rxBuffer[8] << 8) | this->my_rxBuffer[9]) - 30000) / 10;
+    get.packSOC = (float)((this->my_rxBuffer[10] << 8) | this->my_rxBuffer[11]) / 10;
+    return true;
+}
+
+bool Daly_BMS_UART::getMinMaxCellVoltage() //0x91
+{
+    this->sendCommand(COMMAND::MIN_MAX_CELL_VOLTAGE);
+
+    if (!receiveBytes())
+    {
+#ifdef DALY_BMS_DEBUG
+        DEBUG_SERIAL.print("<DALY-BMS DEBUG> Receive failed, min/max cell values won't be modified!\n");
+#endif
+        return false;
+    }
+
+    get.maxCellmV = (float)((this->my_rxBuffer[4] << 8) | this->my_rxBuffer[5]); // Given in mV, convert to V
+    get.maxCellVNum = this->my_rxBuffer[6];
+    get.minCellmV = (float)((this->my_rxBuffer[7] << 8) | this->my_rxBuffer[8]); // Given in mV, convert to V
+    get.minCellVNum = this->my_rxBuffer[9];
 
     return true;
 }
 
-bool Daly_BMS_UART::getPackTemp(int8_t &temp)
+bool Daly_BMS_UART::getPackTemp() //0x92
 {
     this->sendCommand(COMMAND::MIN_MAX_TEMPERATURE);
 
     if (!this->receiveBytes())
     {
 #ifdef DALY_BMS_DEBUG
-        Serial.printf("<DALY-BMS DEBUG> Receive failed, Temp value won't be modified!\n");
+        DEBUG_SERIAL.print("<DALY-BMS DEBUG> Receive failed, Temp value won't be modified!\n");
 #endif
         return false;
     }
 
-    int8_t max_temp = (this->my_rxBuffer[4] - 40);
-    int8_t min_temp = (this->my_rxBuffer[6] - 40);
+    uint8_t max_temp = (this->my_rxBuffer[4] - 40); //byte 0 from datasheet
+    uint8_t min_temp = (this->my_rxBuffer[6] - 40); //byte 3 from datasheet
+    get.tempAverage = (max_temp + min_temp) / 2;
 
-    temp = (max_temp + min_temp) / 2;
+    return true;
+}
+
+bool Daly_BMS_UART::getDischargeChargeMosStatus() //0x93
+{
+    this->sendCommand(COMMAND::DISCHARGE_CHARGE_MOS_STATUS); 
+
+    if (!receiveBytes())
+    {
+#ifdef DALY_BMS_DEBUG
+        DEBUG_SERIAL.print("<DALY-BMS DEBUG> Receive failed, Charge / discharge mos Status won't be modified!\n");
+#endif
+        return false;
+    }
+    get.chargeDischargeStatus = this->my_rxBuffer[4];
+    get.chargeFetState = this->my_rxBuffer[5];
+    get.disChargeFetState = this->my_rxBuffer[6];
+    get.bmsHeartBeat = this->my_rxBuffer[7];
```

Comment:
Could you add a comment here or in the `get` struct explaining what this value means

Reply:
added in struct